### PR TITLE
feat(manager-react-components): upgrade ods version 18

### PR DESCRIPTION
**Pull request by aboungnaseng-ovhcloud**
ref: MANAGER-14622

| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | no
| Breaking change? | no
| Tickets          | #14622
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [x] Breaking change is mentioned in relevant commits

## Description

- upgrade version 18 in manager-react-lib
- update the storybook
- update generator with the new ods version

### CODE REVIEW

**Review by antleblanc — CHANGES_REQUESTED** [review]
Could you please resolve existing conflicts?

File: packages/manager-react-components/src/components/action-banner/action-banner.component.tsx [hunk 1]
```diff
@@ -1,72 +1,68 @@
...
+          />
         )}
         {href && (
-          <OsdsLink
+          <OdsLink
+            data-testid="action-banner-link"
+            id="action-banner-link"
             className="sm:mt-0 mt-4 sm:ml-4 ml-0"
-            color={ODS_THEME_COLOR_INTENT.primary}
+            onClick={() => onClick && onClick()}
```
Comment (chipp972) [inline packages/manager-react-components/src/components/action-banner/action-banner.component.tsx:58]:
```suggestion
            onClick={onClick}
```

File: packages/manager-react-components/src/components/action-banner/action-banner.component.tsx [hunk 2]
```diff
@@ -1,72 +1,68 @@
...
             href={href}
-            target={OdsHTMLAnchorElementTarget._blank}
+            target={'_blank'}
```
Comment (chipp972) [inline packages/manager-react-components/src/components/action-banner/action-banner.component.tsx:60]:
```suggestion
            target="_blank"
```

File: packages/manager-react-components/src/components/action-banner/action-banner.component.tsx [hunk 3]
```diff
@@ -1,72 +1,68 @@
...
 }: Readonly<ActionBannerProps>) {
   return (
-    <OsdsMessage
-      type={type}
-      color={(type as unknown) as ODS_THEME_COLOR_INTENT}
-      className="mt-3 flex-row"
+    <OdsMessage
+      color={color}
+      variant={variant}
+      className={'mt-3 flex-row'}
```
Comment (chipp972) [inline packages/manager-react-components/src/components/action-banner/action-banner.component.tsx:34]:
```suggestion
      className="mt-3 flex-row"
```

File: packages/manager-react-components/src/components/typography/links/links.component.tsx [hunk 4]
```diff
@@ -31,41 +25,14 @@ export const Links: React.FC<LinksProps> = ({
...
-          }
-          hoverable
-          size={ODS_ICON_SIZE.xxs}
-          color={ODS_THEME_COLOR_INTENT.primary}
-        />
-      </span>
-    )}
-  </OsdsLink>
+    {...(type === LinkType.next && { icon: ODS_ICON_NAME.arrowRight })}
+    {...(type === LinkType.external && { icon: ODS_ICON_NAME.externalLink })}
```
Comment (chipp972) [inline packages/manager-react-components/src/components/typography/links/links.component.tsx:33]: We could make an object for this :
```
const iconByLinkType = {
  [LinkType.next]: ODS_ICON_NAME.arrowRight,
  [LinkType.external]: ODS_ICON_NAME.externalLink,
  [LinkType.back]: undefined
}
```

Then in the component :
```
icon={iconByLinkType[type]}
```

File: packages/manager-react-components/src/components/action-banner/action-banner.component.tsx (hunk 1 above)
Comment (aboungnaseng-ovhcloud) [reply packages/manager-react-components/src/components/action-banner/action-banner.component.tsx:58]: 👍

File: packages/manager-react-components/src/components/action-banner/action-banner.component.tsx (hunk 2 above)
Comment (aboungnaseng-ovhcloud) [reply packages/manager-react-components/src/components/action-banner/action-banner.component.tsx:60]: 👍

File: packages/manager-react-components/src/components/action-banner/action-banner.component.tsx (hunk 3 above)
Comment (aboungnaseng-ovhcloud) [reply packages/manager-react-components/src/components/action-banner/action-banner.component.tsx:34]: 👍

File: packages/manager-react-components/src/components/container/step/Step.component.tsx [hunk 5]
```diff
@@ -54,63 +42,62 @@ export const StepComponent = ({
...
-          <OsdsIcon
-            size={ODS_ICON_SIZE.sm}
-            name={ODS_ICON_NAME.CHECK}
-            className={'mr-2'}
-            color={ODS_THEME_COLOR_INTENT.primary}
+          <OdsIcon
+            name={ODS_ICON_NAME.check}
+            className={
+              'block p-[12px] text-[20px] text-[--ods-color-primary-500]'
+            }
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/manager-react-components/src/components/container/step/Step.component.tsx:52]: 👍

File: packages/manager-react-components/src/components/container/step/Step.component.tsx [hunk 6]
```diff
@@ -54,63 +42,62 @@ export const StepComponent = ({
...
           </div>
           {edit?.action && isLocked && (
             <div className="text-2xl w-full md:w-1/6" data-testid="edit">
-              <OsdsLink
+              <OdsButton
+                label={edit.label as string}
                 data-testid="edit-cta"
                 className="float-left md:float-right"
-                color={ODS_THEME_COLOR_INTENT.primary}
                 {...(edit.isDisabled ? { disabled: true } : {})}
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/manager-react-components/src/components/container/step/Step.component.tsx:94]: 👍

File: packages/manager-react-components/src/components/container/step/Step.component.tsx [hunk 7]
```diff
@@ -124,26 +111,22 @@ export const StepComponent = ({
...
+                <OdsButton
                   data-testid="next-cta"
+                  label={next.label as string}
                   size={ODS_BUTTON_SIZE.md}
-                  color={ODS_THEME_COLOR_INTENT.primary}
                   onClick={() => {
                     next.action(id);
                   }}
                   className="w-fit"
                   {...(next.isDisabled ? { disabled: true } : {})}
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/manager-react-components/src/components/container/step/Step.component.tsx:128]: 👍

File: packages/manager-react-components/src/components/content/dashboard-tile/dashboard-tile.component.tsx [hunk 8]
```diff
@@ -26,26 +17,24 @@ export const DashboardTile: React.FC<DashboardTileProps> = ({
...
+          </h4>
+          <OdsDivider spacing="24" />
         </>
       )}
-      {items.map(({ id, label, value }) => (
-        <TileBlock key={id} label={label}>
-          {value}
-        </TileBlock>
+      {items.map((item, index) => (
+        <>
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/manager-react-components/src/components/content/dashboard-tile/dashboard-tile.component.tsx:31]: 👍

File: packages/manager-react-components/src/components/navigation/menus/action/action.component.tsx [hunk 9]
```diff
@@ -45,100 +32,91 @@ export interface ActionMenuProps {
...
+            icon ?? isCompact
+              ? ODS_ICON_NAME.ellipsisHorizontal
+              : ODS_ICON_NAME.chevronDown
+          }
+        />
+      </div>
+      <OdsPopover
+        className="py-[8px] px-0 overflow-hidden"
+        triggerId="navigation-action-trigger"
+        with-arrow="true"
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/manager-react-components/src/components/navigation/menus/action/action.component.tsx:111]: 👍

File: packages/manager-react-components/src/components/templates/delete-modal/delete-modal.component.tsx [hunk 10]
```diff
@@ -60,79 +61,62 @@ export const DeleteModal: React.FC<DeleteModalProps> = ({
...
         {...handleClick(close)}
-      >
-        {cancelButtonLabel || t('deleteModalCancelButton')}
-      </OsdsButton>
-      <OsdsButton
-        disabled={isDisabled}
+        label={cancelButtonLabel || t('deleteModalCancelButton')}
+      />
+      <OdsButton
+        isDisabled={isDisabled}
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/manager-react-components/src/components/templates/delete-modal/delete-modal.component.tsx:111]: 👍

File: packages/manager-react-components/src/components/templates/delete-modal/delete-modal.component.tsx [hunk 11]
```diff
@@ -60,79 +61,62 @@ export const DeleteModal: React.FC<DeleteModalProps> = ({
...
           }
         />
-      </OsdsFormField>
+      </OdsFormField>
       {isLoading && (
         <div className="flex justify-center">
-          <OsdsSpinner inline size={ODS_SPINNER_SIZE.md} />
+          <OdsSpinner size={ODS_SPINNER_SIZE.md} />
         </div>
       )}
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/manager-react-components/src/components/templates/delete-modal/delete-modal.component.tsx:100]: 👍

File: packages/manager-react-components/src/components/templates/update-name-modal/update-name-modal.component.tsx [hunk 12]
```diff
@@ -68,88 +68,68 @@ export const UpdateNameModal: React.FC<UpdateNameModalProps> = ({
...
           </div>
         )}
-      </OsdsFormField>
+      </OdsFormField>
       {isLoading && (
         <div className="flex justify-center">
-          <OsdsSpinner inline size={ODS_SPINNER_SIZE.md} />
+          <OdsSpinner size={ODS_SPINNER_SIZE.md} />
         </div>
       )}
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/manager-react-components/src/components/templates/update-name-modal/update-name-modal.component.tsx:119]: 👍

File: packages/manager-react-components/src/components/templates/update-name-modal/update-name-modal.component.tsx [hunk 13]
```diff
@@ -68,88 +68,68 @@ export const UpdateNameModal: React.FC<UpdateNameModalProps> = ({
...
         {...handleClick(closeModal)}
-      >
-        {cancelButtonLabel || t('updateModalCancelButton')}
-      </OsdsButton>
-      <OsdsButton
-        disabled={isLoading || isPatternError || undefined}
+        label={cancelButtonLabel || t('updateModalCancelButton')}
+      />
+      <OdsButton
+        isDisabled={isLoading || isPatternError}
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/manager-react-components/src/components/templates/update-name-modal/update-name-modal.component.tsx:128]: 👍

File: packages/manager-react-components/src/components/ManagerButton/ManagerButton.tsx [hunk 14]
```diff
@@ -12,45 +8,64 @@ import './translations';
...
-    JSX.OsdsButton &
-      Omit<HTMLAttributes<HTMLOsdsButtonElement>, 'style' | 'id'> &
+    JSX.OdsButton &
+      Omit<HTMLAttributes<HTMLOdsButtonElement>, 'style' | 'id'> &
       StyleReactProps &
-      RefAttributes<HTMLOsdsButtonElement>
+      RefAttributes<HTMLOdsButtonElement>
   >) => {
   const { t } = useTranslation('iam');
   const { isAuthorized } = useAuthorizationIam(iamActions, urn, isIamTrigger);
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/manager-react-components/src/components/ManagerButton/ManagerButton.tsx:36]: yes we can use this isLoading but by default the button is disabled, we have to define the behavior before taking a decision

[the remark this answers is not in the document]
Comment (chipp972) [reply packages/manager-react-components/src/components/ManagerButton/ManagerButton.tsx:36]: I think it is disabled when loading also

File: packages/manager-react-components/src/components/ManagerButton/ManagerButton.tsx [hunk 15]
```diff
@@ -12,45 +8,61 @@ import './translations';
...
-      <OsdsTooltipContent slot="tooltip-content">
+  return displayTooltip ? (
+    <>
+      <div id={id} className="w-fit">
+        <OdsButton
+          {...restProps}
+          data-testid="manager-button-tooltip"
+          isDisabled={true}
+          label={label}
+          content="true"
```
Comment (chipp972) [inline packages/manager-react-components/src/components/ManagerButton/ManagerButton.tsx:51]: What does content="true" do on OdsButton ?

**Review by chipp972 — DISMISSED** [review]

File: packages/manager-react-components/src/components/ManagerButton/ManagerButton.tsx (hunk 15 above)
Comment (aboungnaseng-ovhcloud) [reply packages/manager-react-components/src/components/ManagerButton/ManagerButton.tsx:51]: deleted

File: packages/manager-react-components/src/components/ManagerButton/ManagerButton.tsx [hunk 16]
```diff
@@ -12,45 +8,64 @@ import './translations';
...
       StyleReactProps &
-      RefAttributes<HTMLOsdsButtonElement>
+      RefAttributes<HTMLOdsButtonElement>
   >) => {
   const { t } = useTranslation('iam');
   const { isAuthorized } = useAuthorizationIam(iamActions, urn, isIamTrigger);
 
-  if (isAuthorized) {
-    return <OsdsButton {...restProps}>{children}</OsdsButton>;
+  if (!isAuthorized) {
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/manager-react-components/src/components/ManagerButton/ManagerButton.tsx]: 👍

File: packages/manager-react-components/src/components/ManagerButton/ManagerButton.tsx (hunk 15 above)
Comment (aboungnaseng-ovhcloud) [reply packages/manager-react-components/src/components/ManagerButton/ManagerButton.tsx:51]: 👍

File: packages/manager-react-components/src/components/typography/links/links.component.tsx [hunk 17]
```diff
@@ -31,41 +25,14 @@ export const Links: React.FC<LinksProps> = ({
   type,
   ...props
 }: LinksProps) => (
-  <OsdsLink
-    color={ODS_THEME_COLOR_INTENT.primary}
+  <OdsLink
+    href="#"
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/manager-react-components/src/components/typography/links/links.component.tsx:29]: 👍

File: packages/manager-react-components/src/components/navigation/card/card.component.tsx [hunk 18]
```diff
@@ -61,80 +47,53 @@ export const Card: React.FC<CardProps> = ({
 
   return (
     <a
-      target={OdsHTMLAnchorElementTarget._blank}
+      target={'_blank'}
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/manager-react-components/src/components/navigation/card/card.component.tsx:50]: 👍

**Review by chipp972 — APPROVED** [review]

**Review by darsene — APPROVED** [review]

**Pull request merged**